oci/spec_opts.go

```diff
@@ -636,14 +637,28 @@ func WithUsername(username string) SpecOpts {
 
 // WithAdditionalGIDs sets the OCI spec's additionalGids array to any additional groups listed
 // for a particular user in the /etc/groups file of the image's root filesystem
-func WithAdditionalGIDs(username string) SpecOpts {
+// The passed in user can be either a uid or a username.
+func WithAdditionalGIDs(userstr string) SpecOpts {
 	return func(ctx context.Context, client Client, c *containers.Container, s *Spec) (err error) {
 		setProcess(s)
-		if c.Snapshotter == "" && c.SnapshotKey == "" {
-			if !isRootfsAbs(s.Root.Path) {
-				return errors.Errorf("rootfs absolute path is required")
+		setAdditionalGids := func(root string) error {
+			var username string
+			uid, err := strconv.Atoi(userstr)
+			if err == nil {
+				user, err := getUserFromPath(root, func(u user.User) bool {
+					return u.Uid == uid
+				})
+				if err != nil {
+					if os.IsNotExist(err) || err == errNoUsersFound {
```

Comment:
We should ignore not found error here, because it is normal that a built-from scratch image doesn't have `/etc/passwd`.